src/FpsCounter.cpp

```diff
@@ -14,44 +14,27 @@
 
 using namespace NAS2D;
 
-const unsigned int FPS_COUNTS_SIZE = 25;
-unsigned int FPS_COUNTS[FPS_COUNTS_SIZE] = { 0 };
-
-unsigned int	CURRENT_TICK		= 0;
-unsigned int	LAST_TICK			= 0;
-unsigned int	TICK_DELTA			= 0;
-unsigned int	INDEX				= 0;
-unsigned int	ACCUMULATOR			= 0;
-
-
-/**
- * FpsCounter c'tor
- */
-FpsCounter::FpsCounter()
-{}
-
 
 /**
  * Gets an average count of Frames per Second.
  */
 unsigned int FpsCounter::fps()
 {
-	LAST_TICK = CURRENT_TICK;
-	CURRENT_TICK = SDL_GetTicks();
-
-	TICK_DELTA = CURRENT_TICK - LAST_TICK;
+	auto lastTick = currentTick;
+	currentTick = SDL_GetTicks();
 
-	if (TICK_DELTA == 0) { TICK_DELTA = 1; }
+	auto tickDelta = currentTick - lastTick;
+	if (tickDelta == 0) { tickDelta = 1; }
 
-	FPS_COUNTS[++INDEX] = 1000 / TICK_DELTA;
+	fpsCounts[++fpsCountIndex] = 1000 / tickDelta;
 
-	if (INDEX >= FPS_COUNTS_SIZE) { INDEX = 0; }
+	if (fpsCountIndex >= FpsCountsSize) { fpsCountIndex = 0; }
 
-	ACCUMULATOR = 0;
-	for (auto i : FPS_COUNTS)
+	unsigned int accumulator = 0;
+	for (auto i : fpsCounts)
 	{
-		ACCUMULATOR += i;
+		accumulator += i;
 	}
 
-	return ACCUMULATOR / FPS_COUNTS_SIZE;
+	return accumulator / FpsCountsSize;
```

Comment:
Refactor this to: `std::accumulate(std::begin(fpsCounts), std::end(fpsCounts), 0) / FpsCountsSize;`

`std::reduce` is faster, but I'm not sure of it's adoption yet due to the `std::execution::par` requirements.

Reply:
I was considering using a standard library algorithm here, but wanted to defer on that. The syntax I saw for working examples was less clear the the original code here.

Reply:
This day and age, `std::accumulate` is a well known algorithm. It clearly means "Add these up", `std::reduce` on the other hand: "reduce these to a single value using this function" is quote-unquote more clear, because it's confusing to use std::accumulate to *multiply* values.